boa/src/builtins/object/internal_methods.rs

```diff
@@ -279,43 +271,35 @@ impl Object {
         debug_assert!(Property::is_property_key(prop));
         // Prop could either be a String or Symbol
         match *prop {
-            Value::String(ref st) => {
-                self.properties()
-                    .get(st)
-                    .map_or_else(Property::default, |v| {
-                        let mut d = Property::default();
-                        if v.is_data_descriptor() {
-                            d.value = v.value.clone();
-                            d.writable = v.writable;
-                        } else {
-                            debug_assert!(v.is_accessor_descriptor());
-                            d.get = v.get.clone();
-                            d.set = v.set.clone();
-                        }
-                        d.enumerable = v.enumerable;
-                        d.configurable = v.configurable;
-                        d
-                    })
-            }
+            Value::String(ref st) => self.properties().get(st).map_or_else(Property::empty, |v| {
+                let mut d = Property::empty();
+                if v.is_data_descriptor() {
+                    d.value = v.value.clone();
+                } else {
+                    debug_assert!(v.is_accessor_descriptor());
+                    d.get = v.get.clone();
+                    d.set = v.set.clone();
+                }
+                d.attribute = v.attribute;
+                d
+            }),
             Value::Symbol(ref symbol) => {
                 self.symbol_properties()
                     .get(&symbol.hash())
-                    .map_or_else(Property::default, |v| {
-                        let mut d = Property::default();
+                    .map_or_else(Property::empty, |v| {
+                        let mut d = Property::empty();
                         if v.is_data_descriptor() {
                             d.value = v.value.clone();
-                            d.writable = v.writable;
                         } else {
                             debug_assert!(v.is_accessor_descriptor());
                             d.get = v.get.clone();
                             d.set = v.set.clone();
                         }
-                        d.enumerable = v.enumerable;
-                        d.configurable = v.configurable;
+                        d.attribute = v.attribute;
                         d
                     })
             }
-            _ => Property::default(),
+            _ => unreachable!(""),
```

Comment:
I would add a panic message here, just in case.

Reply:
Yes forgot to add it :sweat_smile: